File: king-admin-java/src/main/java/com/oukingtim/util/ShiroUtils.java
package com.oukingtim.util;

import com.oukingtim.domain.SysUser;
import org.apache.shiro.SecurityUtils;
import org.apache.shiro.session.Session;
import org.apache.shiro.subject.Subject;

/**
 * Created by oukingtim
 */
public class ShiroUtils {

	public static Session getSession() {
		return SecurityUtils.getSubject().getSession();
	}

	public static Subject getSubject() {
		return SecurityUtils.getSubject();
	}

	public static SysUser getUser() {
		return (SysUser) SecurityUtils.getSubject().getPrincipal();
	}

	public static String getUserId() {
		return getUser().getId();
	}
	
	public static void setSessionAttribute(Object key, Object value) {
		getSession().setAttribute(key, value);
	}

	public static Object getSessionAttribute(Object key) {
		return getSession().getAttribute(key);
	}

	public static boolean isLogin() {
		return SecurityUtils.getSubject().getPrincipal() != null;
	}

	public static void logout() {
		SecurityUtils.getSubject().logout();
	}
	
	public static String getKaptcha(String key) {
		String kaptcha = getSessionAttribute(key).toString();
		getSession().removeAttribute(key);
		return kaptcha;
	}

}

File: king-admin-kotlin/src/main/kotlin/com/oukingtim/util/ShiroUtils.kt
package com.oukingtim.util

import com.oukingtim.domain.SysUser
import org.apache.shiro.SecurityUtils
import org.apache.shiro.session.Session
import org.apache.shiro.subject.Subject

/**
 * Created by oukingtim
 */
object ShiroUtils {

    fun getSession(): Session {
        return SecurityUtils.getSubject().session
    }

    fun getSubject(): Subject {
        return SecurityUtils.getSubject()
    }

    fun getUser(): SysUser {
        return SecurityUtils.getSubject().principal as SysUser
    }

    fun getUserId(): String {
        return getUser().id!!
    }

    fun setSessionAttribute(key: Any, value: Any) {
        getSession().setAttribute(key, value)
    }

    fun getSessionAttribute(key: Any): Any {
        return getSession().getAttribute(key)
    }

    fun isLogin(): Boolean {
        return SecurityUtils.getSubject().principal != null
    }

    fun logout() {
        SecurityUtils.getSubject().logout()
    }

    fun getKaptcha(key: String): String {
        val kaptcha = getSessionAttribute(key).toString()
        getSession().removeAttribute(key)
        return kaptcha
    }
}